Initial MSQ Templates

OutKa5t opened the pull request:
Added template file for MSQL
Added test and parameter file for MSQL
Added initial reeadme
Added NSG template for MSQL

Ticket NCSD-1029

OutKa5t committed:
Initial MSQ Templates

Added template file for MSQL
Added test and parameter file for MSQL
Added initial reeadme
Added NSG template for MSQL

Ticket NCSD-1029

OutKa5t committed:
Update azure-pipelines.yml

Added specific pool.

rivoric commented on README.md:
```diff
@@ -0,0 +1,9 @@
+# dfc-shared
```
It's not dfc-shared

OutKa5t replied:
Fixed

rivoric commented on README.md:
```diff
@@ -0,0 +1,9 @@
+# dfc-shared
+
+[![Build status](https://sfa-gov-uk.visualstudio.com/Digital%20First%20Careers/_apis/build/status/DFC%20Shared/dfc-shared?branchName=master)](https://sfa-gov-uk.visualstudio.com/Digital%20First%20Careers/_build/latest?definitionId=1690)
```
Link to dfc-shared build

OutKa5t replied:
Fixed.

rivoric reviewed (commented):
Readme still referencing dfc-shared

rivoric commented on template.json:
```diff
@@ -0,0 +1,171 @@
+      "name": "sharedManagedInstanceEnvironment",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2017-05-10",
+      "condition": "[equals(parameters('envType'),'dev')]",
```
I thought you needed this deploying to preprod? If so you can remove the envType parameter completely

OutKa5t replied:
Dropped

rivoric commented on template.json:
```diff
@@ -0,0 +1,171 @@
+    {
+      "name": "managedSQLNSG",
+      "type": "Microsoft.Resources/deployments",
+      "condition": "[parameters('deployManageSql')]",
```
Is there ever a case where you are going to run this and not want to deploy the managed SQL? Assuming not then you can completely remove the deployManageSql param

OutKa5t replied:
Maybe, if the network needs to change.

rivoric commented on parameters.json:
```diff
@@ -0,0 +1,51 @@
+    "deployManageSql": {
+      "value": "__deployManageSql__"
+    },
+    "msqlResourceGroupName": {
```
msqlResourceGroupName is not used anywhere

OutKa5t replied:
Dropped

rivoric commented on test-parameters.json:
```diff
@@ -0,0 +1,52 @@
+    "deployManageSql": {
+      "value": true
+    },
+    "msqlResourceGroupName": {
```
msqlResourceGroupName is not used anywhere

rivoric commented on test-parameters.json:
```diff
@@ -0,0 +1,52 @@
+    "msqlSubnetName": {
+      "value": "dfc-local-msql"
+    },
+    "msqlNetworkSecurityGroup": {
```
msqlNetworkSecurityGroup is not used anywhere

rivoric commented on parameters.json:
```diff
@@ -0,0 +1,51 @@
+    "msqlSubnetName": {
+      "value": "__msqlSubnetName__"
+    },
+    "msqlNetworkSecurityGroup": {
```
msqlNetworkSecurityGroup

rivoric commented on parameters.json:
```diff
@@ -0,0 +1,51 @@
+    "virtualNetworkNonRoutedSubnets": {
+      "value": "__virtualNetworkNonRoutedSubnets__"
+    },
+    "threatDetectionEmailAddress": {
```
threatDetectionEmailAddress is not used anywhere

rivoric commented on test-parameters.json:
```diff
@@ -0,0 +1,52 @@
+        "dfc-local-nrs"
+      ]
+    },
+    "threatDetectionEmailAddress": {
```
threatDetectionEmailAddress is not used anywhere

OutKa5t replied:
Removed

OutKa5t committed:
Update parameters.json

Removed extra param not in use.

OutKa5t committed:
Update template.json

Removed condition on networking but kept condition on sql deployment to be able to update the network without the long wait

OutKa5t committed:
Remove unwanted params

Tody up parameters and naming.

OutKa5t committed:
Update parameters.json

Changed format of parameter arrays to accomodate tokenised value and valid schema

OutKa5t committed:
Update template.json

Renamed the network deployment.

OutKa5t committed:
Updated readme

Corrected the link to the build status.

OutKa5t committed:
Updated readme

Corrected the title.

OutKa5t committed:
Folder structure update

Matching the rest of DFC repos using 'Resources' folder.
